File: leetcode/0088-merge-sorted-array/java/solution_2.java
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        
        if (n == 0) {
            return;
        }
        
        int[] ans = new int[m+n];
    
        int i = 0;
        int j = 0;
        int k = 0;
        while (i < m && j < n) {
            
            if (nums1[i] <= nums2[j]) {
                ans[k++] = nums1[i];
                i++;
            } else {
                ans[k++] = nums2[j];
                j++;
            }
        }
        
        while(i < m) {
            ans[k++] = nums1[i];
            i++;
        }

        while(j < n) {
            ans[k++] = nums2[j];
            j++;
        }
        
        for (i=0; i < ans.length; i++) {
            nums1[i] = ans[i];
        }
    }
}

File: leetcode/0199-binary-tree-right-side-view/rust/solution_2.rs
// Definition for a binary tree node.
// #[derive(Debug, PartialEq, Eq)]
// pub struct TreeNode {
//   pub val: i32,
//   pub left: Option<Rc<RefCell<TreeNode>>>,
//   pub right: Option<Rc<RefCell<TreeNode>>>,
// }
// 
// impl TreeNode {
//   #[inline]
//   pub fn new(val: i32) -> Self {
//     TreeNode {
//       val,
//       left: None,
//       right: None
//     }
//   }
// }

// 1
// 2, 3
// 4
// 5
use std::rc::Rc;
use std::cell::RefCell;
use std::collections::VecDeque;

impl Solution {
    pub fn right_side_view(root: Option<Rc<RefCell<TreeNode>>>) -> Vec<i32> {

        let mut ans: Vec<i32> = Vec::new();
        let mut queue: VecDeque<Rc<RefCell<TreeNode>>> = VecDeque::new();

        if let Some(ref node) = root {

            queue.push_back(node.clone());

            while !queue.is_empty() {
                
                ans.push(queue.back().unwrap().borrow().val);
                let len = queue.len();

                for _ in 0..len {
                    let current = queue.pop_front().unwrap().clone();

                    let left_option = current.borrow().left.clone();
                    let right_option = current.borrow().right.clone();

                    if let Some(ref left) = left_option {
                        queue.push_back(left.clone());
                    }
                    if let Some(ref right) = right_option {
                        queue.push_back(right.clone());
                    }
                }
            }
        }
        ans
    }
}

//1
//2, 3
//5,4